crates/deltalake-core/src/logstore/mod.rs

```diff
@@ -37,6 +91,58 @@ lazy_static! {
     static ref DELTA_LOG_PATH: Path = Path::from("_delta_log");
 }
 
+/// Return the [LogStoreRef] for the provided [Url] location
+///
+/// This will use the built-in process global [crate::storage::ObjectStoreRegistry] by default
+///
+/// ```rust
+/// # use deltalake_core::logstore::*;
+/// # use std::collections::HashMap;
+/// # use url::Url;
+/// let location = Url::parse("file:///tmp").expect("Failed to make location");
+/// let logstore = logstore_for(location, HashMap::new()).expect("Failed to get a logstore");
+/// ```
+pub fn logstore_for(
+    location: Url,
+    options: impl Into<StorageOptions> + Clone,
+) -> DeltaResult<LogStoreRef> {
+    // turn location into scheme
+    let scheme = Url::parse(&format!("{}://", location.scheme()))
+        .map_err(|_| DeltaTableError::InvalidTableLocation(location.clone().into()))?;
+
+    if let Some(entry) = crate::storage::factories().get(&scheme) {
+        debug!("Found a storage provider for {scheme} ({location})");
+        let (store, _prefix) = entry
+            .value()
+            .parse_url_opts(&location, &options.clone().into())?;
+        return logstore_with(store, location, options);
+    }
+    Err(DeltaTableError::InvalidTableLocation(
+        location.clone().into(),
+    ))
+}
+
+/// Return the [LogStoreRef] using the given [ObjectStoreRef]
+pub fn logstore_with(
+    store: ObjectStoreRef,
+    location: Url,
+    options: impl Into<StorageOptions> + Clone,
+) -> DeltaResult<LogStoreRef> {
+    let scheme = Url::parse(&format!("{}://", location.scheme()))
+        .map_err(|_| DeltaTableError::InvalidTableLocation(location.clone().into()))?;
+
+    if let Some(factory) = logstores().get(&scheme) {
```

Comment:
i think we need to consider both scheme and host in the registries. Otherwise we cannot register stores for differnt buckets with differnet credentials.

The `file:///` stores will always have an empty host, except in some arcane windows cases, that I have not yet encountered in the wild...